include/http/request_handler.js

```diff
@@ -564,6 +576,25 @@ module.exports = function RequestHandlerModule(pb) {
         });
     };
 
+    function checkRouteContainsLocale(url, siteObj){
+        //Do not redirect for public resources
+        if(url.indexOf('/public/') !== -1 || url.indexOf('/admin') !== -1 || url.indexOf('/actions/') !== -1 || url.indexOf('/api') !== -1 || url.indexOf('/media') !== -1) {
+            return false;
+        }
+
+        //Do not redirect if the url contains a supported locale.
+        if(siteObj.supportedLocales) {
+            var loca‌‌les =  Object.keys(siteObj.supportedLocales);
+            for (var i = 0; i < loca‌‌les.length; i++) {
+                if(url.indexOf(loca‌‌les[i]) != -1){
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
```

Comment:
As I mentioned above, I don't think this function is necessary once the route is derived.  You will be able to check the localized property